/*
 * This file was automatically generated by EvoSuite
 * Tue Feb 28 05:35:01 GMT 2017
 */

package org.cs362.dominion;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.cs362.dominion.board;
import org.cs362.dominion.hand;
import org.cs362.dominion.player;
import org.cs362.dominion.salvager;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class salvager_ESTest extends salvager_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      salvager salvager0 = new salvager();
      player player0 = new player();
      board board0 = new board();
      hand hand0 = player0.player_hand;
      hand0.add_card(salvager0);
      hand0.add_card(salvager0);
      hand0.add_card(salvager0);
      player[] playerArray0 = new player[4];
      playerArray0[2] = player0;
      salvager0.perform_attack(playerArray0, 2, board0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      salvager salvager0 = new salvager();
      board board0 = new board();
      // Undeclared exception!
      try { 
        salvager0.perform_attack((player[]) null, 1035, board0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.cs362.dominion.salvager", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      salvager salvager0 = new salvager();
      board board0 = new board();
      player[] playerArray0 = new player[4];
      // Undeclared exception!
      try { 
        salvager0.perform_attack(playerArray0, 3787, board0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 3787
         //
         assertThrownBy("org.cs362.dominion.salvager", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      salvager salvager0 = new salvager();
      player[] playerArray0 = new player[2];
      player player0 = new player();
      playerArray0[0] = player0;
      board board0 = new board();
      salvager0.perform_attack(playerArray0, 0, board0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      salvager salvager0 = new salvager();
      player player0 = new player();
      board board0 = new board();
      hand hand0 = player0.player_hand;
      hand0.add_card(salvager0);
      hand0.add_card(salvager0);
      player[] playerArray0 = new player[4];
      playerArray0[2] = player0;
      salvager0.perform_attack(playerArray0, 2, board0);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      salvager salvager0 = new salvager();
      salvager0.description();
  }
}
